src/main/kotlin/parts/wisdom/arcraider/visualization/GridPane.kt

```diff
@@ -0,0 +1,42 @@
+package parts.wisdom.arcraider.visualization
+
+import java.awt.Color
+import java.awt.Graphics
+import javax.swing.JPanel
+
+// When calculating the width / height of each grid pixel, offset it so
+// we don't fill right to the edge of the screen.
+private const val OFFSET = 10
+
+class GridPane(val grid: Grid) : JPanel() {
+    override fun paintComponent(g: Graphics) {
+        super.paintComponent(g)
+        background = Color.BLACK
+        val graphics2d = g.create()
+        graphics2d.color = Color.WHITE
+
+        val xSize = (width - OFFSET) /  grid.width
+        val ySize = (height - OFFSET) / grid.height
+        val xStart = (getWidth() - xSize * grid.width) / 2
+        val yStart = (getHeight() - ySize * grid.height) / 2
+
+        var y = yStart
+
+        // Draw the Empty Grid.
+        for (vert in 0 until grid.height) {
```

Comment:
Personally I'd compute `x` and `y` rather than incrementally maintaining them. But I think that's a matter of personal taste.

Reply:
fixed